exporter/stats/stackdriver/stackdriver.go

```diff
@@ -309,3 +322,30 @@ func namespacedViewName(v string, escaped bool) string {
 	}
 	return path.Join("custom.googleapis.com", p)
 }
+
+func equalAggWindow(md *metricpb.MetricDescriptor, agg stats.Aggregation, window stats.Window) bool {
+	var w stats.Window
+	var a stats.Aggregation
+
+	switch md.MetricKind {
+	case metricpb.MetricDescriptor_DELTA:
+		w = &stats.Interval{}
+	case metricpb.MetricDescriptor_CUMULATIVE:
+		w = &stats.Cumulative{}
+	}
+
+	switch md.ValueType {
+	case metricpb.MetricDescriptor_INT64:
+		a = &stats.CountAggregation{}
+	case metricpb.MetricDescriptor_DISTRIBUTION:
+		a = &stats.DistributionAggregation{}
+	}
+
+	if reflect.TypeOf(agg) != reflect.TypeOf(a) {
```

Comment:
We should also check whether agg is a pointer or not, or disallow concrete types to be passed as aggregators.